mirrord/cli/src/internal_proxy.rs

```diff
@@ -103,7 +122,7 @@ pub(crate) async fn proxy() -> Result<()> {
                 let agent_connection = connect(&config).await?;
                 active_connections.spawn(connection_task(stream, agent_connection));
             },
-            _ = active_connections.join_next() => {},
+            Some(_) = active_connections.join_next(), if !active_connections.is_empty() => {},
             _ = tokio::time::sleep(Duration::from_secs(1)) => {
```

Comment:
Im not sure how this will behave because the timeout will always resolve so anything that is None for active_connections and Err for listener is swollen and I think